airbyte-config/models/src/test/java/io/airbyte/config/helpers/KubeLoggingConfigTest.java

```diff
@@ -31,7 +31,8 @@
   public void cleanUpLogs() {
     if (logPath != null) {
       try {
-        LogClientSingleton.deleteLogs(new EnvConfigs(), logPath);
+        final EnvConfigs envConfigs = new EnvConfigs();
+        LogClientSingleton.getInstance().deleteLogs(envConfigs.getWorkerEnvironment(), envConfigs.getLogConfigs(), logPath);
```

Comment:
A static helper isn't quite the same as a singleton, and can be much harder to control and initialize.  Using a constructor or getInstance() pattern ensures that it's initialized in a controlled manner, instead of silently hoping that the calling code sets up configuration properly before using the methods that depend on state.  A singleton instance provides a single point of entry for initialization, and thus stronger guarantees around how something is configured before being used.  It's a stronger contract.  This is why there's a difference between static state holding a singleton instance, vs static state breaking encapsulation by exposed properties for anyone to change.

Reply:
See my other comments. I was trying to go for a static class here to avoid any sort of initialisation so the singleton name is my mistake.